### PR TITLE
feat(governance): bridge UDT on ARB and OP chains

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds a way to deploy a bridged token to Optimism chains, as well as a script to bridge some tokens.

- [x] deploy bridged erc20 on ARB
 - [x] deploy bridged erc20 on OP
 - [x] deploy bridged erc20 on Base
 - [x] deploy bridged erc20 on Base Sepolia
 - [x] test token bridge on ARB
 - [x]  test token bridge on OP
 - [x]  test token bridge on Base Sepolia
 - [ ]  test token bridge on Base

# Issues

Refs https://github.com/unlock-protocol/unlock/issues/11676

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: governance/contracts/l2/UDTOptimism.sol
Comment (clemsos) [inline governance/contracts/l2/UDTOptimism.sol:1]: For now this contract is stored in `governance` folder but should be moved to smart-contracts. We have conflicts in ethers version as optimism SDK requires ethers5 and out deploy helpers requires ethers6...

File: smart-contracts/scripts/l2tokens.js
Comment (clemsos) [inline smart-contracts/scripts/l2tokens.js:1]: Same thing. This should go into `governance` folder. We should be able to solve the sdk ethers5 issue with yarn `resolutions` but I hadnt been able too so far https://github.com/ethereum-optimism/optimism/issues/6295

**Review by julien51 — APPROVED** [review]

File: smart-contracts/scripts/l2tokens.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const OP_SEPOLIA_NETWORK = {
+  name: 'Op Sepolia',
+  id: 11155420,
+  provider: 'https://sepolia.optimism.io',
+}
+
+async function main({
+  l1TokenAddress = L1_UDT_SEPOLIA,
+  l2TokenAddress = L2_UDT_OP_SEPOLIA,
+  l1ChainId = 11155111, // Sepolia
```
[the remark this answers is not in the document]
Comment (clemsos) [reply smart-contracts/scripts/l2tokens.js:20]: yes. We dont have OP Sepolia in it. I was going to add Base Sepolia in networks package so maybe use that as default after

File: governance/contracts/l2/UDTOptimism.sol [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    _mint(_to, _amount);
+    emit Mint(_to, _amount);
+  }
+
+  /// @notice Prevents tokens from being withdrawn to L1.
+  function burn(
+    address,
+    uint256
+  ) external virtual override(IOptimismMintableERC20) onlyBridge {
+    revert("UnlockDiscountToken cannot be withdrawn");
```
Comment (clemsos) [inline governance/contracts/l2/UDTOptimism.sol:113]: this is one question we need to address

Comment (julien51) [reply governance/contracts/l2/UDTOptimism.sol:113]: Hum, Please clarify. I think it should be withdrawn! What's blocking here?

**Comment by clemsos** [comment]
It happens that UDT was already bridged on Arbitrum... https://arbiscan.io/address/0xd5d3aA404D7562d09a848F96a8a8d5D65977bF90#readProxyContract

**Comment by clemsos** [comment]
I am struggling to bridge the tokens on Base, because of gas limit/price that is not correct. We are running into out of gas error like here https://etherscan.io/tx/0x74b08c2475fb299b380fded66d4c285139c192e394468db8a4ad3f841a6ccbfa

It works on simulation with exactly same parameters but different gas settings, also works fine on Base Sepolia which mimicks the exact same token / bridge setup.

**Comment by julien51** [comment]
Ok let me try to get do it as well!

**Comment by julien51** [comment]
I used the official UIs to make some tests:

* [X] [Optimism](https://app.optimism.io/bridge/deposit ) [L1 deposit]( https://etherscan.io/tx/0x4dd4c7d1bc18a0f1d78fef17cdf0d18e96380c51c096fe0994ba5fa7083dde47), [L2 incoming](https://optimistic.etherscan.io/tx/0x56f0055fa364dd03164bed51dffb8aeb4187f3b43911b4bdcedec137c661fde7)
* [ ] [Arbitrum](https://bridge.arbitrum.io/?destinationChain=arbitrum-one&sourceChain=ethereum) [L1 deposit](), [L2 incoming]()
* [ ] [Base](https://bridge.base.org/deposit)  [L1 deposit](), [L2 incoming]()

**Comment by julien51** [comment]
https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/759

**Review by julien51 — APPROVED** [review]
LGTM!

**Pull request merged**